### ISSUE TITLE
Network module shows 100% signal strength regardless of actual strength

### ISSUE THREAD

**Issue by t3zla** [enkore/i3pystatus#201]
My WiFi signal strength is not 100% while the network module claims it is

**Comment by t3zla** [comment]
Actually, it does not show 100% all the time, it shows whatever the signal strength was when the computer boots up. It never updates

**Comment by gacekjk** [comment]
Same issue here, but imo it seems that it's the problem with the wireless drivers rather than the module. It seems that the information from the card is not updated to the system (I've no idea how the network manager is obtaining it).
Please check if the information in /proc/net/wireless is updated on your system (on mine it's not).
Intel wireless card?

**Comment by enkore** [comment]
basiciw is a pretty simple module and just asks libiw what it guesses the current ESSID and signal levels are ... so yeah, there's not much going on and not much potential for bugs like this.

So I'd blame the drivers, too ;)

**Comment by terminalmage** [comment]
Yeah, some drivers are better than others. The broadcom drivers are particularly bad in this respect.

**Comment by richese** [comment]
Speaking of proprietary broadcoms drivers, this might be related:
My signal quality is always 0 and I get these two errors every `interval`  seconds on my pc with broadcom wifi card:
```
mar 29 17:33:19 yip kernel: ERROR @wl_cfg80211_get_tx_power : error (-1)
mar 29 17:33:19 yip kernel: ERROR @wl_dev_intvar_get : error (-1)
```
I had to comment a few lines in [basiciw](https://github.com/enkore/basiciw/blob/985b24dbc157c23b97330b0eef6f9d95660b96e0/basiciw.c#L87) to get rid of the constant error messages.
The same errors show up when running `iwconfig` command.

**Comment by enkore** [comment]
I see these too on one of my laptops with a bcm chip. The error message doesn't really say what it doesn't like, probably the driver just doesn't have these variables.
